src/offset/mod.rs

```diff
@@ -209,10 +209,119 @@ pub trait Offset: Sized + Clone + fmt::Debug {
     fn fix(&self) -> FixedOffset;
 }
 
-/// The time zone.
+/// The [`TimeZone`] trait is the backbone of chrono, allowing it to work with local time.
 ///
-/// The methods here are the primarily constructors for [`Date`](../struct.Date.html) and
-/// [`DateTime`](../struct.DateTime.html) types.
+/// # Using `TimeZone`
+///
+/// Most methods of this trait are deprecated, it is usually better or easier to work with the
+/// methods on [`DateTime`] and [`NaiveDateTime`] instead.
+///
+/// A type implementing [`TimeZone`] can be used to create a [`DateTime`]:
+///
+/// Creating a [`DateTime`] from a [`NaiveDateTime`]:
+/// - [`from_local_datetime`](#method.from_local_datetime) if the datetime is local.
+/// - [`from_utc_datetime`](#method.from_utc_datetime) if the datetime is in utc.
+///
+/// Creating a [`DateTime`] from a Unix timestamp in UTC:
+/// - [`timestamp_opt`](#method.timestamp_opt)
+/// - [`timestamp_millis_opt`](#method.timestamp_millis_opt)
+/// - [`timestamp_nanos`](#method.timestamp_nanos)
+///
+/// Creating a [`DateTime`] from year, month, day, time components:
+/// - [`with_ymd_and_hms`](#method.with_ymd_and_hms)
+///
+/// Parsing a [`DateTime`] from a string:
+/// - [`datetime_from_str`](#method.datetime_from_str). Note that if the string contains an offset
+///   from UTC, that offset must match the timezone at that point in time.
+///
+/// # Implementing `TimeZone`
+///
+/// The primary methods are [`offset_from_local_datetime`](#tymethod.offset_from_local_datetime) and
+/// [`offset_from_utc_datetime`](#tymethod.offset_from_utc_datetime), which calculate the offset
+/// from UTC that applies at a specific datetime. That specific datetime can
+/// be given as either local or UTC.
+///
+/// [`from_offset`](#tymethod.from_offset) allows the type to be reconstructed from the associated
+/// [`Offset`] type, which gets stored in a [`DateTime`].
+///
+/// ## Example
+///
+#[cfg_attr(not(feature = "clock"), doc = "```ignore")]
+#[cfg_attr(feature = "clock", doc = "```rust")]
+/// use chrono::{LocalResult, Offset, TimeZone};
```

Comment:
Why not one `use` statement?

Reply:
I had some reasoning... But if I can't remember it must not have been that good :smile:.